### PR TITLE
graphQl-903: deprecated use_for_shipping in billing address schema

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
# Fixed Issues (if relevant)
1. #903: [Checkout] Replace use_for_shipping with same_as_shipping

### CODE REVIEW

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 1]
```diff
@@ -96,7 +96,8 @@ input SetBillingAddressOnCartInput {
 input BillingAddressInput {
     customer_address_id: Int
     address: CartAddressInput
-    use_for_shipping: Boolean
+    use_for_shipping: Boolean @doc(description: "Deprecated. Use same_as_shipping")
+    same_as_shipping: Boolean
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls]: Please add `@doc` here.

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 2]
```diff
@@ -96,7 +96,8 @@ input SetBillingAddressOnCartInput {
 input BillingAddressInput {
     customer_address_id: Int
     address: CartAddressInput
-    use_for_shipping: Boolean
+    use_for_shipping: Boolean @doc(description: "Deprecated. Use same_as_shipping")
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls]:
```suggestion
    use_for_shipping: Boolean @doc(description: "Deprecated: use `same_as_shipping` field instead")
```

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-5963](https://jira.corp.magento.com/browse/ENGCOM-5963) has been created to process this Pull Request
:eight_spoked_asterisk: @lenaorobei, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Despite the fact that `use_for_shipping` is deprecated we need to preserve its behaviour. It should be the same as `same_as_shipping`.

File: app/code/Magento/QuoteGraphQl/Model/Cart/SetBillingAddressOnCart.php [hunk 3]
```diff
@@ -58,6 +58,8 @@ public function execute(ContextInterface $context, CartInterface $cart, array $b
         $addressInput = $billingAddressInput['address'] ?? null;
         $useForShipping = isset($billingAddressInput['use_for_shipping'])
             ? (bool)$billingAddressInput['use_for_shipping'] : false;
+        $sameAsShipping = isset($billingAddressInput['same_as_shipping'])
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/Model/Cart/SetBillingAddressOnCart.php:71]: We need to keep `use_for_shipping` for BC. So, if `same_as_shipping ` is not specified, but `use_for_shipping` the code should still work as expected. But if both are specified - `same_as_shipping` should be preferred.

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/SetBillingAddressOnCartTest.php [hunk 4]
```diff
@@ -142,7 +142,7 @@ public function testSetNewBillingAddressWithUseForShippingParameter()
           telephone: "88776655"
           save_in_address_book: false
          }
-         use_for_shipping: true
```
Comment (lenaorobei) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/SetBillingAddressOnCartTest.php]: We should keep tests for both fields for now.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please see my comment about BC.

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-5963](https://jira.corp.magento.com/browse/ENGCOM-5963) has been created to process this Pull Request
:eight_spoked_asterisk: @lenaorobei, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Pull request merged**